### ISSUE TITLE
Make it easier to opt-in existing Python projects to use automatic virtual environment support

### ISSUE THREAD

**Issue by justinvp** [pulumi/pulumi#4777]
With https://github.com/pulumi/pulumi/issues/4645, _new_ Python projects will automatically have a virtual environment created, dependencies installed, and `Pulumi.yaml` updated to specify the `virtualenv` runtime option, so that it is used automatically by Pulumi.

As a follow-up to that, we should consider making it easier to opt-in _existing_ projects that don't already have a virtual environment created. For example, when you git clone the examples repo and are trying to get started with an existing Python example.

Basically, having some gesture that can be used with existing projects to easily create the virtual environment, install dependencies, and update `Pulumi.yaml` to save the `virtualenv` runtime option.

This could be either a new CLI command (e.g. something like `pulumi prepare` or `pulumi install`) or interactive prompt to an existing command (e.g. `pulumi stack init`).

**Comment by justinvp** [comment]
Today, the steps to do this manually are:

Linux/macOS:

```sh
$ python3 -m venv venv
$ venv\bin\pip install -r requirements.txt
```

Windows:

```batch
> python -m venv venv
> venv/Scripts/pip.exe install -r requirements.txt
```

---

Then update `Pulumi.yaml`, replacing `runtime: python` with:

```yaml
runtime:
  name: python
  options:
    virtualenv: venv
```

**Comment by justinvp** [comment]
Of course, if we made the creation of the virtual environment automatic when `virtualenv` is set and the specified dir doesn't exist per #5164, then this is less of a concern and we could just go and update all our existing examples to have the `virtualenv` option set.
